raco/language/myrialang.py

```diff
@@ -566,6 +570,9 @@ def __init__(self, input):
     def num_tuples(self):
         return self.input.num_tuples()
 
+    def partitioning(self):
```

Comment:
I can certainly understand this argument; however I prefer to keep it this way for two reasons:
1. `partitioning()` is a statement about the output of an operator. Either the Consumer or the Producer (or both) must conceptually be responsible for getting partitioned tuples to the destined worker. MyriaX never has anything between a MyriaShuffleProducer and MyriaShuffleConsumer. So, it is valid to say either op starts the partitioned output.
2. So which should we choose? ops like `MyriaShuffleProducer` are the ones in the Producer/Consumer pair that have information about shuffle attributes (notice `MyriaShuffleConsumer` has no members beyond those of `UnaryOperator`), so it is much simpler to allow them to calculate it